src/librustc/ty/mod.rs

```diff
@@ -2296,8 +2296,8 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
         self.get_attrs(did).iter().any(|item| item.check_name(attr))
     }
 
-    pub fn trait_has_default_impl(self, trait_def_id: DefId) -> bool {
-        self.trait_def(trait_def_id).has_default_impl
+    pub fn trait_has_auto_impl(self, trait_def_id: DefId) -> bool {
```

Comment:
Can we rename this to `trait_is_auto`? The concept of an `auto_impl` is on the chopping block, after all.

Reply:
Also, can we add a doc comment like this:

```rust
/// Returns true if this is an `auto trait`.
///
/// NB. For a limited time, also returns true if `impl Trait for .. { }` is in the code-base.
```